### ISSUE TITLE
Writing to the same file from multiple stencils causes malformed output

### ISSUE THREAD

**Issue by Cyberbeni** [krzysztofzablocki/Sourcery#881]
I wanted to put init and isEqual and hash generation to 3 seperate files but once the `// sourcery:inline:auto:...` was generated for the class I was testing it with, the next code generation cut off the end of the first one in the file, the start of the last one and both ends of the middle one. If I put them all in 1 stencil, they are working correctly.

When having 2 inline blocks automatically generated, enabling a 3rd one can still cause the newly generated inline block to overlap with a previous one

**Comment by krzysztofzablocki** [comment]
Generally we recommend using inline code-gen only if it's required by the language for multitude of reasons but merge conflicts with other teammates might get annoying if you do this too often. 

The way `:auto` works is that it places the code at the end of class declaration, so if you are going to use multiple templates then it makes sense it's messed up because we don't reparse file, so if you are going to do something like this you can try not using `:auto` variant but explicitly specyfing the markers with proper name

**Comment by Cyberbeni** [comment]
Isn't it higher chance to get merge conflict if all the generated code is in 1 file?

One of our use case is to generate code for plain model classes (init, isEqual, etc.), so having the generated functions next to the variables in this case is more readable for us.

Since having the stencils all in one file works mostly good, adding a flag to handle the input stencils as a single unit should be a good quick fix.

**Comment by krzysztofzablocki** [comment]
with normal.generated files merge conflicts aren't a problem because you simply run Sourcery again and it replaces the content.

> should be a good quick fix
happy to accept that pr :)

**Comment by Cyberbeni** [comment]
As there is an easy workaround (manually merge the stencils), it is pretty low on my TODO list.

And I guess calling sourcery separately within the build script is a better workaround as it allows using separate files (A little bit slower but luckily sourcery is pretty fast). So something like this:
```bash
for stencil in $(find "path/to/stencils" -name "*.stencil"); do
    sourcery ... --templates "$stencil"
done
```
I haven't had time to test it, will do so later, but I'm pretty sure this will work. If you have a "has workaround" label, I think you can tag this issue with it as I don't think anyone will take the time to improve this use case anytime soon when the workaround is almost as good.

**Comment by Cyberbeni** [comment]
We commit files as you suggested in the video linked in the README and we are also running UI tests on macOS 10.11 and the brew formula says xcode 12.0 required, so I don't think not commiting the generated files would work for us. (Only the test runner has to be rebuilt but that's where a lot of the generated code is needed for Swift 3.0 support)

**Comment by krzysztofzablocki** [comment]
I think the better workaround for this limitation is my original suggestion of using explicit markers not ':auto'

**Comment by Cyberbeni** [comment]
fixed by #944
